src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Import/ImportOrchestratorJobTests.cs

```diff
@@ -68,16 +86,561 @@ public async Task GivenAnOrchestratorJobAndWrongEtag_WhenOrchestratorJobStart_Th
+
+            ImportOrchestratorJobErrorResult resultDetails = (ImportOrchestratorJobErrorResult)jobExecutionException.Error;
+            Assert.Equal(HttpStatusCode.BadRequest, resultDetails.HttpStatusCode);
+            Assert.Equal(2, testQueueClient.JobInfos.Count(t => t.Status == JobStatus.Completed));
+            Assert.Equal(1, testQueueClient.JobInfos.Count(t => t.Status == JobStatus.Failed));
+            Assert.Equal(2, testQueueClient.JobInfos.Count(t => t.Status == JobStatus.Cancelled));
+
+            _ = mediator.Received().Publish(
+               Arg.Is<ImportJobMetricsNotification>(
+                   notification => notification.Id == orchestratorJobInfo.Id.ToString() &&
+                   notification.Status == JobStatus.Failed.ToString() &&
+                   notification.CreatedTime == importOrchestratorInputData.CreateTime &&
+                   notification.SucceedCount == 1 &&
+                   notification.FailedCount == 1),
+               Arg.Any<CancellationToken>());
+        }
+
+        [Fact]
+        public async Task GivenAnOrchestratorJob_WhenLastSubJobCancelld_ThenOperationCanceledExceptionShouldBeThrowAndWaitForOtherSubJobsCompleted()
+        {
+            IImportOrchestratorJobDataStoreOperation fhirDataBulkImportOperation = Substitute.For<IImportOrchestratorJobDataStoreOperation>();
+            RequestContextAccessor<IFhirRequestContext> contextAccessor = Substitute.For<RequestContextAccessor<IFhirRequestContext>>();
+            ILoggerFactory loggerFactory = new NullLoggerFactory();
+            IIntegrationDataStoreClient integrationDataStoreClient = Substitute.For<IIntegrationDataStoreClient>();
+            ISequenceIdGenerator<long> sequenceIdGenerator = Substitute.For<ISequenceIdGenerator<long>>();
+            IMediator mediator = Substitute.For<IMediator>();
+            ImportOrchestratorJobInputData importOrchestratorInputData = new ImportOrchestratorJobInputData();
+            ImportOrchestratorJobResult importOrchestratorJobResult = new ImportOrchestratorJobResult();
+            TestQueueClient testQueueClient = new TestQueueClient();
+            int getJobByGroupIdCalledTime = 0;
+            testQueueClient.GetJobByIdFunc = (queueClient, id, _) =>
+            {
+                JobInfo jobInfo = queueClient.JobInfos.First(t => t.Id == id);
+                if (jobInfo.Id == 3)
+                {
+                    jobInfo.Status = JobStatus.Cancelled;
+                    jobInfo.Result = JsonConvert.SerializeObject(new ImportProcessingJobErrorResult() { Message = "Job Cancelled." });
+                }
+
+                return jobInfo;
+            };
+            testQueueClient.GetJobByGroupIdFunc = (queueClient, groupId, _) =>
+            {
+                IEnumerable<JobInfo> jobInfos = queueClient.JobInfos.Where(t => t.GroupId == groupId);
+                if (++getJobByGroupIdCalledTime > 2)
```

Comment:
>2 [](http://example.com/codeflow?start=50&length=1)

why here is 2?

Reply:
Because I want to ensure the function WaitCancelledJobCompletedAsync in ImportOrchestratorJob.cs can finish after three calls.